<a href="https://colab.research.google.com/github/Mel-iza/Data-Science-e-Machine-Learning-Tera/blob/main/Notebooks%20/Model/Modelos_de_Classifica%C3%A7%C3%A3o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd 

Express Pre-processing - merge data

In [2]:
# falso
#telegram_data = pd.read_csv('/content/telegram_data_preprocessed___.csv')

# verificar split
veritas = pd.read_csv('/content/veritas_preprocessed__.csv')
data_factck = pd.read_csv('/content/new_data_factck_preprocessed__ (1).csv')

# dados verdadeiros
fake_online_corpus = pd.read_csv('/content/fake_online_corpus_true_preprocessed_.csv')
fake_br_corpus_nilc = pd.read_csv('/content/fake_br_corpus_nilc__.csv')


In [9]:
fake_online_corpus['label'].unique()

array(['verdadeiro'], dtype=object)

In [51]:
telegram_data['label'] = 'falso'
print('false',telegram_data.shape)

false (17987, 7)


In [3]:
print(fake_br_corpus_nilc.shape)
print(fake_br_corpus_nilc['label'].unique())

(3600, 3)
[ True]


In [4]:
veritas_true = veritas[veritas['label']=='verdadeiro']
veritas_false = veritas[veritas['label']=='falso']

print(veritas_true['label'].unique(),veritas_true.shape)
print(veritas_false['label'].unique(),veritas_false.shape)

['verdadeiro'] (3263, 5)
['falso'] (2224, 5)


In [5]:
fake_online_corpus_true = fake_online_corpus[fake_online_corpus['label']=='verdadeiro']
fake_online_corpus_false = fake_online_corpus[fake_online_corpus['label']=='falso']

print('true',fake_online_corpus_true.shape)
print('false',fake_online_corpus_false.shape)

true (1630, 7)
false (0, 7)


In [6]:
data_factck_true = data_factck[data_factck['label']=='verdadeiro']
data_factck_false = data_factck[data_factck['label']=='falso']

print('true', data_factck_true.shape)
print('false', data_factck_false.shape)

true (129, 8)
false (1166, 8)



|Dataset|valor mensagem| Quantidade colunas|Quantidade linhas|
|-------|--------------|-------------------|-----------------|
|telegram|falso|7|17987|
|veritas|verdadeiro|5|3263|
|veritas|falso|5|2224|
|fake online corpus|verdadeiro|7|1630|
|Factck br corpus|verdadeiro|8|129|
|Factck br corpus|falso|8|1166|
|Fake br corpus nilc|verdadeiro|3|3600|

<br>

||Total Raw|
|-----|-------------|
|Verdadeiros|8.622|
|Falsos|21.377|

<br>

||Total Pós Zero-shot|
|-----|-------------|
|Verdadeiros|8.622|
|Falsos|15.901|

In [ ]:
grupos_para_merge = fake_br_corpus_nilc, veritas_true, fake_online_corpus, data_factck_true

novo df = ['text', 'label', 'fonte'] <br>

**fake_nilc** = preprocessed_news, label, fonte <br>
**veritas** = noticia, label, fonte <br>
**fake_online_corpus** = text, label, fonte <br>
**data_factck** = text, label, fonte


In [ ]:
fake_online_corpus fake_br_corpus_nilc veritas

In [25]:
fake_br_corpus_nilc['fonte'] = 'Fake.Br Corpus NILC'

In [73]:
veritas

url  \
0     https://chequeado.com/verificacionfb/no-nostra...   
1     https://piaui.folha.uol.com.br/lupa/2021/02/05...   
2     https://maldita.es/malditobulo/2020/04/03/ayud...   
3     https://piaui.folha.uol.com.br/lupa/2021/03/25...   
4     https://politica.estadao.com.br/blogs/estadao-...   
...                                                 ...   
5482  https://economia.uol.com.br/noticias/redacao/2...   
5483  https://www.camara.leg.br/noticias/680401-CAMA...   
5484  https://noticias.uol.com.br/ultimas-noticias/e...   
5485  https://www.camara.leg.br/noticias/647043-PROJ...   
5486  https://checamos.afp.com/esta-cena-de-um-hospi...   

                                                 titulo  \
0     nao ha registro da suposta profecia no livro d...   
1               verificamos variante coronavirus manaus   
2     o site acessado nao oferece a possibilidade de...   
3                   verificamos inalacao agua sanitaria   
4     nao ha evidencias de que nostradamus tenha esc...   
...                                                 ...   
5482  guedes diz que pode derrubar encargos trabalhi...   
5483  camara aprova mp que destinou r bilhoes para a...   
5484  espanha soma mil infectados pela covid em um d...   
5485  projeto preve alcool em gel gratuito para grup...   
5486  esta cena de um hospital apos um incendio foi ...   

                                                noticia       label    fonte  
0                      nostradamus previu o coronavirus       falso  veritas  
1     a nova variante do sarscov e mais letal e nao ...       falso  veritas  
2     mensagem que mostra um link que leva a um site...       falso  veritas  
3     inalacao de bicarbonato de sodio com agua sani...       falso  veritas  
4     postagens de uma profecia de nostradamus sobre...       falso  veritas  
...                                                 ...         ...      ...  
5482  segundo ele a medida seria tomada la na frente...  verdadeiro  veritas  
5483    najara araujocamara dos deputados  deputados...  verdadeiro  veritas  
5484  o pais acordou hoje com ruas vazias lojas fech...  verdadeiro  veritas  
5485       compartilhe      versao para impressao   ...  verdadeiro  veritas  
5486  uma foto de pessoas deitadas em macas e aglome...  verdadeiro  veritas  

[5487 rows x 5 columns]

In [ ]:
veritas_true_new = veritas_true.rename({'noticia':'text'}, axis=1)


In [29]:
fake_online_corpus['fonte'] = 'FakeOnlineTCC-Corpus'

In [38]:
data_factck_true['fonte'] = 'FACTCK.BR'

<ipython-input-38-b85fd33a626e>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_factck_true['fonte'] = 'FACTCK.BR'


In [40]:
data_factck_true.sample(3)

título  \
443   lvaro dias acerta ao falar sobre queda dos inv...   
928   verificamos publicacao lista acoes de bolsonar...   
1168  verificamosenem checamos ao vivo informacoes q...   

                                                   text  \
443   lvaro dias podemos afirmou no seu plano de gov...   
928                                                       
1168                                                      

                                          claimReviewed   ano  mês      fonte  \
443   o investimento nacional chegou em a do pib par...  2018    8  FACTCK.BR   
928   a bolsa de valores superou mil pontos pela pri...  2019    4  FACTCK.BR   
1168  o enem tira ponto quando acha que eu chutei a ...  2018   11  FACTCK.BR   

                                                    url       label  
443   https://apublica.org/2018/08/truco-alvaro-dias...  verdadeiro  
928   https://piaui.folha.uol.com.br/lupa/2019/04/17...  verdadeiro  
1168  https://piaui.folha.uol.com.br/lupa/2018/11/03...  verdadeiro

In [55]:
data_factck_true.value_counts()

título                                                                                                 text                                                                                                                                                                                                                                          claimReviewed                                                                                                                                     ano   mês  fonte      url                                                                                                            label     
 frases dos presidenciaveis sobre seguranca publica agencia publica                                    jair bolsonaro psl afirmou que nos estados unidos o numero de mortes por mil habitantes e na ordem de cinco vezes menos do que aqui                                                                                                           nos estados unidos o numero 

In [56]:
data_factck_true_space = data_factck_true[data_factck_true['text']!=' ']


In [57]:
data_factck_true_space

título  \
349  papa francisco nao enviou terco a lula vatican...   
363  em economia bolsonaro cita dados falsos e hadd...   
375  verdades e mentiras misturamse em corrente con...   
376  verdades e mentiras misturamse em corrente con...   
377  verdades e mentiras misturamse em corrente con...   
..                                                 ...   
739   frases que mostram o poder de cunha agencia p...   
744  checamos frases sobre impeachment e planos de ...   
753  checamos argumentos pro e contra o impeachment...   
756  checamos argumentos pro e contra o impeachment...   
758  checamos argumentos pro e contra o impeachment...   

                                                  text  \
349  como informou nota do vaticano publicada nesta...   
363  fernando haddad pt em entrevista no roda viva ...   
375  corrente que circula no whatsapp diz que hadda...   
376  corrente que circula no whatsapp diz que hadda...   
377  corrente que circula no whatsapp diz que hadda...   
..                                                 ...   
739  o deputado federal pompeo de mattos pdtrs afir...   
744  o deputado federal paulo pimenta ptrs afirmou ...   
753  miguel reale junior jurista e coautor do pedid...   
756  jose eduardo martins cardozo ministro da advoc...   
758  jose eduardo martins cardozo ministro da advoc...   

                                         claimReviewed   ano  mês      fonte  \
349     papa envia terco a lula preso politico ha dias  2018    6  FACTCK.BR   
363  hoje a classificacao do brasil e superior a da...  2018   10  FACTCK.BR   
375  haddad e reu por improbidade em acao que apura...  2018   10  FACTCK.BR   
376  haddad e acusado de enriquecimento ilicito por...  2018   10  FACTCK.BR   
377  addad e acusado de improbidade em acao que inv...  2018   10  FACTCK.BR   
..                                                 ...   ...  ...        ...   
739  esses que estao pedindo a saida do deputado wa...  2016    5  FACTCK.BR   
744  encontrase nesta casa um pedido de impeachment...  2016    4  FACTCK.BR   
753  juizo de admissibilidade e apenas o reconhecim...  2016    4  FACTCK.BR   
756  a decisao do presidente eduardo cunha nao viso...  2016    4  FACTCK.BR   
758  todo o foco da denuncia diz respeito a situaca...  2016    4  FACTCK.BR   

                                                   url       label  
349  https://aosfatos.org/noticias/papa-francisco-n...  verdadeiro  
363  https://apublica.org/2018/10/truco-em-economia...  verdadeiro  
375  https://apublica.org/2018/10/truco-verdades-e-...  verdadeiro  
376  https://apublica.org/2018/10/truco-verdades-e-...  verdadeiro  
377  https://apublica.org/2018/10/truco-verdades-e-...  verdadeiro  
..                                                 ...         ...  
739  https://apublica.org/2016/05/truco-3-frases-qu...  verdadeiro  
744  https://apublica.org/2016/04/truco-checamos-4-...  verdadeiro  
753  https://apublica.org/2016/04/truco-checamos-12...  verdadeiro  
756  https://apublica.org/2016/04/truco-checamos-12...  verdadeiro  
758  https://apublica.org/2016/04/truco-checamos-12...  verdadeiro  

[86 rows x 8 columns]

In [5]:
veritas

url  \
0   https://chequeado.com/verificacionfb/no-nostra...   
1   https://piaui.folha.uol.com.br/lupa/2021/02/05...   
2   https://maldita.es/malditobulo/2020/04/03/ayud...   
3   https://piaui.folha.uol.com.br/lupa/2021/03/25...   
4   https://politica.estadao.com.br/blogs/estadao-...   
..                                                ...   
95  https://www.boatos.org/saude/covid-19-farsa-cr...   
96  https://www.boatos.org/politica/festa-casa-do-...   
97  https://piaui.folha.uol.com.br/lupa/2021/02/01...   
98  https://piaui.folha.uol.com.br/lupa/2021/02/08...   
99  https://piaui.folha.uol.com.br/lupa/2020/10/20...   

                                               titulo  \
0   nao ha registro da suposta profecia no livro d...   
1             verificamos variante coronavirus manaus   
2   o site acessado nao oferece a possibilidade de...   
3                 verificamos inalacao agua sanitaria   
4   nao ha evidencias de que nostradamus tenha esc...   
..                                                ...   
95  covid e uma farsa criada para venda de orgaos ...   
96  festa na casa do filho de joao doria e realiza...   
97  verificamos hidroxicloroquina estados unidos t...   
98  os dois remedios eram parte do tratamento padr...   
99               verificamos nanorrobos vacinas covid   

                                              noticia  label    fonte  
0                    nostradamus previu o coronavirus  falso  veritas  
1   a nova variante do sarscov e mais letal e nao ...  falso  veritas  
2   mensagem que mostra um link que leva a um site...  falso  veritas  
3   inalacao de bicarbonato de sodio com agua sani...  falso  veritas  
4   postagens de uma profecia de nostradamus sobre...  falso  veritas  
..                                                ...    ...      ...  
95  entao vou mandar o eu vou mandar o texto para ...  falso  veritas  
96  versao filho de joao doria da festa em mansao ...  falso  veritas  
97  instituicoes cientificas americanas passaram a...  falso  veritas  
98  estudo da usp comprovou que hidroxicloroquina ...  falso  veritas  
99  vacinas contem nanorobos que podem alterar o d...  falso  veritas  

[100 rows x 5 columns]

---

```
novo df = ['text', 'label', 'fonte']

fake_nilc = preprocessed_news, label, fonte 
veritas = noticia, label, fonte
fake_online_corpus = text, label, fonte
data_factck = text, label, fonte
```

*data_factck_true_space* <br> fake_online_corpus 
 <br>  fake_br_corpus_nilc <br> veritas

In [68]:
import pandas as pd

colunas_selecionadas = ['text', 'label', 'fonte']

def selecionar_colunas(dataframe, colunas_selecionadas):
    novo_dataframe = dataframe[colunas_selecionadas]
    return novo_dataframe

In [79]:
datasets_for_merge = [data_factck_true_space, fake_online_corpus, fake_br_corpus_nilc_new, veritas_true_new]

In [80]:
# Lista de colunas selecionadas
colunas_selecionadas = ['text', 'label', 'fonte']

# Lista para armazenar os novos dataframes
novos_datasets = []

# Aplicar a função em cada dataset
for dataset in datasets_for_merge:
    novo_dataframe = selecionar_colunas(dataset, colunas_selecionadas)
    novos_datasets.append(novo_dataframe)

# Imprimir os novos datasets
for novo_dataset in novos_datasets:
    print(novo_dataset)
    print("------------------------")


                                                  text       label      fonte
349  como informou nota do vaticano publicada nesta...  verdadeiro  FACTCK.BR
363  fernando haddad pt em entrevista no roda viva ...  verdadeiro  FACTCK.BR
375  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
376  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
377  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
..                                                 ...         ...        ...
739  o deputado federal pompeo de mattos pdtrs afir...  verdadeiro  FACTCK.BR
744  o deputado federal paulo pimenta ptrs afirmou ...  verdadeiro  FACTCK.BR
753  miguel reale junior jurista e coautor do pedid...  verdadeiro  FACTCK.BR
756  jose eduardo martins cardozo ministro da advoc...  verdadeiro  FACTCK.BR
758  jose eduardo martins cardozo ministro da advoc...  verdadeiro  FACTCK.BR

[86 rows x 3 columns]
------------------------
                

In [89]:
novos_datasets[3]

text       label    fonte
2224  enquanto isso o sxsw foi cancelado em austin n...  verdadeiro  veritas
2225   de abril de  h sorocaba com recordes de morte...  verdadeiro  veritas
2226  publicacoes compartilhadas milhares de vezes e...  verdadeiro  veritas
2227       atualizado em             o plenario da c...  verdadeiro  veritas
2228    ione morenoprefeitura de manaus  lei aprovad...  verdadeiro  veritas
...                                                 ...         ...      ...
5482  segundo ele a medida seria tomada la na frente...  verdadeiro  veritas
5483    najara araujocamara dos deputados  deputados...  verdadeiro  veritas
5484  o pais acordou hoje com ruas vazias lojas fech...  verdadeiro  veritas
5485       compartilhe      versao para impressao   ...  verdadeiro  veritas
5486  uma foto de pessoas deitadas em macas e aglome...  verdadeiro  veritas

[3263 rows x 3 columns]

In [107]:
datasets_true_merged = pd.concat([novos_datasets[0], novos_datasets[1], novos_datasets[2], novos_datasets[3]])
datasets_true_merged

text       label      fonte
349   como informou nota do vaticano publicada nesta...  verdadeiro  FACTCK.BR
363   fernando haddad pt em entrevista no roda viva ...  verdadeiro  FACTCK.BR
375   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
376   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
377   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
...                                                 ...         ...        ...
5482  segundo ele a medida seria tomada la na frente...  verdadeiro    veritas
5483    najara araujocamara dos deputados  deputados...  verdadeiro    veritas
5484  o pais acordou hoje com ruas vazias lojas fech...  verdadeiro    veritas
5485       compartilhe      versao para impressao   ...  verdadeiro    veritas
5486  uma foto de pessoas deitadas em macas e aglome...  verdadeiro    veritas

[8579 rows x 3 columns]

In [108]:
datasets_true_merged['label'].unique()

array(['verdadeiro', True], dtype=object)

In [109]:
datasets_true_merged['label'] = datasets_true_merged['label'].astype(str)

In [110]:
datasets_true_merged['label'].unique()

array(['verdadeiro', 'True'], dtype=object)

In [111]:
datasets_true_merged['label'] = datasets_true_merged['label'].replace(['True'], 'verdadeiro')
datasets_true_merged['label'].unique()

array(['verdadeiro'], dtype=object)

In [112]:
datasets_true_merged

text       label      fonte
349   como informou nota do vaticano publicada nesta...  verdadeiro  FACTCK.BR
363   fernando haddad pt em entrevista no roda viva ...  verdadeiro  FACTCK.BR
375   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
376   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
377   corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
...                                                 ...         ...        ...
5482  segundo ele a medida seria tomada la na frente...  verdadeiro    veritas
5483    najara araujocamara dos deputados  deputados...  verdadeiro    veritas
5484  o pais acordou hoje com ruas vazias lojas fech...  verdadeiro    veritas
5485       compartilhe      versao para impressao   ...  verdadeiro    veritas
5486  uma foto de pessoas deitadas em macas e aglome...  verdadeiro    veritas

[8579 rows x 3 columns]

In [113]:
from google.colab import files

datasets_true_merged.to_csv('datasets_true_merged_.csv', header=True, index=False, encoding='utf-8')
files.download('datasets_true_merged_.csv') 

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:

novo_dataframe = selecionar_colunas(data_factck_true_space, colunas_selecionadas)
novo_dataframe

In [ ]:
colunas_selecionadas = ['text', 'label', 'fonte']
for_merge_data_factck = data_factck_true_space[colunas_selecionadas]
for_merge_data_factck

In [66]:
colunas_selecionadas = ['text', 'label', 'fonte']
for_merge_data_factck = data_factck_true_space[colunas_selecionadas]
for_merge_data_factck

text       label      fonte
349  como informou nota do vaticano publicada nesta...  verdadeiro  FACTCK.BR
363  fernando haddad pt em entrevista no roda viva ...  verdadeiro  FACTCK.BR
375  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
376  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
377  corrente que circula no whatsapp diz que hadda...  verdadeiro  FACTCK.BR
..                                                 ...         ...        ...
739  o deputado federal pompeo de mattos pdtrs afir...  verdadeiro  FACTCK.BR
744  o deputado federal paulo pimenta ptrs afirmou ...  verdadeiro  FACTCK.BR
753  miguel reale junior jurista e coautor do pedid...  verdadeiro  FACTCK.BR
756  jose eduardo martins cardozo ministro da advoc...  verdadeiro  FACTCK.BR
758  jose eduardo martins cardozo ministro da advoc...  verdadeiro  FACTCK.BR

[86 rows x 3 columns]

---

In [6]:
!pip install transformers

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.1/7.1 MB 42.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 236.8/236.8 kB 26.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.8/7.8 MB 100.6 MB/s eta 0:00:00


In [14]:
from transformers import pipeline
import pandas as pd

df = pd.DataFrame()

classifier = pipeline("zero-shot-classification",
                      model='Mel-Iza0/zero-shot', tokenizer='Mel-Iza0/zero-shot')

candidate_labels = ['vacina',
                    'vacinação',
                    'imunização',
                    'ciência',
                    'hobbie',
                    'culinária',
                    'compras']
hypothesis_template = "Este exemplo é sobre {}."

results = []
for data in veritas['noticia'].iterrows:
    classification = classifier(data, candidate_labels, hypothesis=hypothesis_template)['labels']
    results.append(classification[-1])

df['model'] = results


Failed to determine 'entailment' label id from the label2id mapping in the model config. Setting to -1. Define a descriptive label2id mapping in the model config to ensure correct outputs.
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.


In [32]:
from transformers import pipeline
import pandas as pd


classifier = pipeline("zero-shot-classification",
                      model='Mel-Iza0/zero-shot', tokenizer='Mel-Iza0/zero-shot')

candidate_labels = ['vacina',
                    'vacinação',
                    'imunização',
                    'ciência',
                    'hobbie',
                    'culinária',
                    'compras']
hypothesis_template = "Este exemplo é sobre {}."


for i, row in veritas.iterrows():
    noticia = veritas.loc[i, 'noticia']
    veritas.at[i, 'classificacao'] = classifier(noticia, candidate_labels, hypothesis=hypothesis_template)['labels'][-1]

veritas


Failed to determine 'entailment' label id from the label2id mapping in the model config. Setting to -1. Define a descriptive label2id mapping in the model config to ensure correct outputs.
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.


                                                  url  \
0   https://chequeado.com/verificacionfb/no-nostra...   
1   https://piaui.folha.uol.com.br/lupa/2021/02/05...   
2   https://maldita.es/malditobulo/2020/04/03/ayud...   
3   https://piaui.folha.uol.com.br/lupa/2021/03/25...   
4   https://politica.estadao.com.br/blogs/estadao-...   
..                                                ...   
95  https://www.boatos.org/saude/covid-19-farsa-cr...   
96  https://www.boatos.org/politica/festa-casa-do-...   
97  https://piaui.folha.uol.com.br/lupa/2021/02/01...   
98  https://piaui.folha.uol.com.br/lupa/2021/02/08...   
99  https://piaui.folha.uol.com.br/lupa/2020/10/20...   

                                               titulo  \
0   nao ha registro da suposta profecia no livro d...   
1             verificamos variante coronavirus manaus   
2   o site acessado nao oferece a possibilidade de...   
3                 verificamos inalacao agua sanitaria   
4   nao ha evidencias de que n

In [33]:
veritas

url  \
0   https://chequeado.com/verificacionfb/no-nostra...   
1   https://piaui.folha.uol.com.br/lupa/2021/02/05...   
2   https://maldita.es/malditobulo/2020/04/03/ayud...   
3   https://piaui.folha.uol.com.br/lupa/2021/03/25...   
4   https://politica.estadao.com.br/blogs/estadao-...   
..                                                ...   
95  https://www.boatos.org/saude/covid-19-farsa-cr...   
96  https://www.boatos.org/politica/festa-casa-do-...   
97  https://piaui.folha.uol.com.br/lupa/2021/02/01...   
98  https://piaui.folha.uol.com.br/lupa/2021/02/08...   
99  https://piaui.folha.uol.com.br/lupa/2020/10/20...   

                                               titulo  \
0   nao ha registro da suposta profecia no livro d...   
1             verificamos variante coronavirus manaus   
2   o site acessado nao oferece a possibilidade de...   
3                 verificamos inalacao agua sanitaria   
4   nao ha evidencias de que nostradamus tenha esc...   
..                                                ...   
95  covid e uma farsa criada para venda de orgaos ...   
96  festa na casa do filho de joao doria e realiza...   
97  verificamos hidroxicloroquina estados unidos t...   
98  os dois remedios eram parte do tratamento padr...   
99               verificamos nanorrobos vacinas covid   

                                              noticia  label    fonte  \
0                    nostradamus previu o coronavirus  falso  veritas   
1   a nova variante do sarscov e mais letal e nao ...  falso  veritas   
2   mensagem que mostra um link que leva a um site...  falso  veritas   
3   inalacao de bicarbonato de sodio com agua sani...  falso  veritas   
4   postagens de uma profecia de nostradamus sobre...  falso  veritas   
..                                                ...    ...      ...   
95  entao vou mandar o eu vou mandar o texto para ...  falso  veritas   
96  versao filho de joao doria da festa em mansao ...  falso  veritas   
97  instituicoes cientificas americanas passaram a...  falso  veritas   
98  estudo da usp comprovou que hidroxicloroquina ...  falso  veritas   
99  vacinas contem nanorobos que podem alterar o d...  falso  veritas   

   results_zero_shot classificacao  
0         imunização    imunização  
1             vacina        vacina  
2          culinária     culinária  
3         imunização    imunização  
4          culinária     culinária  
..               ...           ...  
95            vacina        vacina  
96         culinária     culinária  
97           ciência       ciência  
98           compras       compras  
99         vacinação     vacinação  

[100 rows x 7 columns]

In [15]:
df.value_counts()

model     
imunização    27
vacina        22
vacinação     22
culinária     11
compras        8
ciência        6
hobbie         4
dtype: int64

In [18]:
veritas['results_zero_shot'] = results

In [19]:
veritas

url  \
0   https://chequeado.com/verificacionfb/no-nostra...   
1   https://piaui.folha.uol.com.br/lupa/2021/02/05...   
2   https://maldita.es/malditobulo/2020/04/03/ayud...   
3   https://piaui.folha.uol.com.br/lupa/2021/03/25...   
4   https://politica.estadao.com.br/blogs/estadao-...   
..                                                ...   
95  https://www.boatos.org/saude/covid-19-farsa-cr...   
96  https://www.boatos.org/politica/festa-casa-do-...   
97  https://piaui.folha.uol.com.br/lupa/2021/02/01...   
98  https://piaui.folha.uol.com.br/lupa/2021/02/08...   
99  https://piaui.folha.uol.com.br/lupa/2020/10/20...   

                                               titulo  \
0   nao ha registro da suposta profecia no livro d...   
1             verificamos variante coronavirus manaus   
2   o site acessado nao oferece a possibilidade de...   
3                 verificamos inalacao agua sanitaria   
4   nao ha evidencias de que nostradamus tenha esc...   
..                                                ...   
95  covid e uma farsa criada para venda de orgaos ...   
96  festa na casa do filho de joao doria e realiza...   
97  verificamos hidroxicloroquina estados unidos t...   
98  os dois remedios eram parte do tratamento padr...   
99               verificamos nanorrobos vacinas covid   

                                              noticia  label    fonte  \
0                    nostradamus previu o coronavirus  falso  veritas   
1   a nova variante do sarscov e mais letal e nao ...  falso  veritas   
2   mensagem que mostra um link que leva a um site...  falso  veritas   
3   inalacao de bicarbonato de sodio com agua sani...  falso  veritas   
4   postagens de uma profecia de nostradamus sobre...  falso  veritas   
..                                                ...    ...      ...   
95  entao vou mandar o eu vou mandar o texto para ...  falso  veritas   
96  versao filho de joao doria da festa em mansao ...  falso  veritas   
97  instituicoes cientificas americanas passaram a...  falso  veritas   
98  estudo da usp comprovou que hidroxicloroquina ...  falso  veritas   
99  vacinas contem nanorobos que podem alterar o d...  falso  veritas   

   results_zero_shot  
0         imunização  
1             vacina  
2          culinária  
3         imunização  
4          culinária  
..               ...  
95            vacina  
96         culinária  
97           ciência  
98           compras  
99         vacinação  

[100 rows x 6 columns]

In [ ]:
print('Acertos: {} de {}'.format(len(df.loc[df.true_class == df[model]]), len(df)))
print('Erros: {} de {}'.format(len(df.loc[df.true_class != df[model]]), len(df)))

In [ ]:
from transformers import pipeline

df = pd.DataFrame()

for data in veritas['noticia']:
  classifier = pipeline("zero-shot-classification",
  model='Mel-Iza0/zero-shot', tokenizer='Mel-Iza0/zero-shot')

  candidate_labels = ['vacina',
                      'vacinação',
                      'imunização',
                      'ciência',
                      'entretenimento',
                      'culinária','compras']
  hypothesis_template = "Este exemplo é sobre {}."
  
  results = []
  results.append(data['noticia'].apply(lambda x: classifier(x, candidate_labels, hypothesis=hypothesis_template)['labels'][-1]))
  data['model'] = results


In [ ]:
from transformers import pipeline

df = pd.DataFrame()

classifier = pipeline("zero-shot-classification",
model='Mel-Iza0/zero-shot', tokenizer='Mel-Iza0/zero-shot')

candidate_labels = ['vacina', 'vacinação', 'imunização', 'entretenimento', 'política']
hypothesis_template = "Este exemplo é sobre {}."

results = []
results.append(data['preprocessed_news'].apply(lambda x: classifier(x, candidate_labels, hypothesis=hypothesis_template)['labels'][-1]))
print(results)